File: base-ios/MyCare/Model/City.swift
//
//  City.swift
//  MyCare
//
//  Created by Cookie on 4/16/20.
//  Copyright © Mohd Nasir Selamat. All rights reserved.
//

import Foundation

struct CityRequest: Encodable {
    var stateCode: String
    var deviceInfo: DeviceInfo
    
    enum CodingKeys: String, CodingKey {
        case stateCode = "stateCd"
    }
}

struct AllCities: Decodable {
    var id: Int
    var code: String
    var description: String
    
    enum CodingKeys: String, CodingKey {
        case id = "cityId"
        case code = "cityCd"
        case description = "cityDesc"
    }
}

File: angular/base-bff-api/projects/bff-api/src/lib/models/city.ts
import { State } from './state';

export class City {
    cityId?: number;
    cityCd?: string;
    cityDesc?: string;
    state?: State;

    constructor(args: {
        cityId?: number,
        cityCd?: string,
        cityDesc?: string,
        state?: State
    }) {
        this.cityId = args.cityId;
        this.cityCd = args.cityCd;
        this.cityDesc = args.cityDesc;
        this.state = args.state;
    }
}
